### PR TITLE
Provide an iterator over all the documents in a milli index

### PR DESCRIPTION
**Pull request by irevoire**

### CODE REVIEW

File: milli/src/index.rs [hunk 1]
```diff
@@ -477,6 +477,18 @@ impl Index {
...
+    /// Returns an iterator over all the documents in the index.
+    pub fn all_documents<'t>(
+        &self,
+        rtxn: &'t RoTxn,
+    ) -> anyhow::Result<impl Iterator<Item = heed::Result<(DocumentId, obkv::KvReader<'t>)>>> {
+        Ok(self
+            .documents
+            .iter(rtxn)?
+            // we cast the BEU32 to a DocumentId
+            .map(|document| document.map(|(id, obkv)| (id.get(), obkv))))
```
Comment (Kerollmops) [inline milli/src/index.rs:489]:
```suggestion
        // we cast the BEU32 to a DocumentId
        Ok(self.documents.iter(rtxn)?.map(|result| result.map(|(id, obkv)| (id.get(), obkv))))
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Yeah, that's great, thank you!

File: milli/src/index.rs (hunk 1 above)
Comment (irevoire) [reply milli/src/index.rs:489]: are you sure you want to do that? RustFmt is not happy with this change 🥺

**Review by Kerollmops — APPROVED** [review]
Thank you, RustFmt is better than me at formatting :)

bors merge

**Pull request merged**